requirements.txt

```diff
@@ -24,6 +24,4 @@ progressbar2
 -f https://download.pytorch.org/whl/torch_stable.html
 torch==1.7.1+cu110
 torchvision==0.8.2+cu110
-https://download.lfd.uci.edu/pythonlibs/q4trcu4l/GDAL-3.2.3-cp38-cp38-win_amd64.whl  # Python 3.8 and Win 10 64 Bit needed
-https://download.lfd.uci.edu/pythonlibs/q4trcu4l/Fiona-1.8.19-cp38-cp38-win_amd64.whl # Python 3.8 and Win 10 64 Bit needed
-geopandas
\ No newline at end of file
+geopandas
```

Comment:
@martinbaerwolff, for the test and the gui `gdal` and `fiona` did not have to be installed here. Actually, `fiona` is installed anyway, probably as a requirement of a third-party package. In general I noticed that there are several packages listed, that are not explicitly imported anywhere within OTVision (if I checked correctly). If these are requirements of third-party packages, shouldn't their installation handle it?

Another point: as far as I saw it, `torch` on pip is on version 1.9, but you explicitly require 1.7 here. Is that necessary?

Reply:
Actually, when I used `pyreqs` to determine the dependencies to be put in the requirements file, this is what I would get:

```py
geopandas>=0.9.0
opencv_python>=4.5.2.54
torch>=1.9.0
pandas>=1.2.5
numpy>=1.21.0
matplotlib>=3.4.2
pause>=0.3
Pillow>=8.2.0
progressbar33>=2.4
PySimpleGUI>=4.45.0
```

Reply:
but it seems that there are some dependencies of `torch` that are not properly resolved, e.g. `requests`  or `yaml`. At least I get an error when detecting road users in the GUI when I don't add it to the dependencies above.

Reply:
Hey @dglaeser, thanks for your commits!

Actually we didn't maintain all the requirements.txt files... whoops ;)

Definitely something we should work on. Maybe it's the appropriate moment to do it.

Installing `gdal` and `fiona` with pip on windows is a mess... (conda works well). It's much easier on linux. The easiest workaround is to include the whole url in the requirements. But it won't work on linux, that's true. Our main focus is on windows for the moment. A dirty workaround for the moment could be a separate `requirements-linux.txt`.

Reply:
ah I see. I had no problem installing e.g. `gdal` on Linux, but I also seemed to not need it - at least for the road user detection and the `api_test`. I didn't check all funcitonality available via the GUI. I suppose you need it at some point for coordinate transformation from image coordinates to geographic? As a side note, I personally find the `pyproj` library easier to use (which is used by `gdal` under the hood as far as I know). But if you need `gdal` for instance for I/O to raster formats or similar, then it may not be an alternative anyway..

Reply:
Not quite sure... @martinbaerwolff ?

I think @martinbaerwolff just tested a `geopandas` Dataframe (which depends on fiona which depends on gdal) for internal handling and export as geopackage.

Reply:
so apparently they recommend to place all, even indirectly used packages with the exact versions (`==`-syntax) in `requirements.txt` files. I think this is very useful in cases where you want to share an exact snapshot (e.g. in case you write a paper using `OTVision` and you want to provide the source code running the examples)...

If you'd like `OTVision` to be usable in a framework sense, we could think of adding a `setup.py` file that contains only the abstract dependencies, and allows system-wide package installation (and possibly hosting it on pypi).

Reply:
Totally with you!

setup.py and pypi distribution is already on the todo list :)

Reply:
The reason we added `gdal` and `fiona` to the requirements is just because of the hazzle installing it without the provided url.

Reply:
I am totally with all the arguments mentioned by both of you. I use geopandas only for writing tracks in world coords to gpkg and as @frunika said on Windows pip installation of geopandas errors out when automatically installing GDAL and fiona as dependencies. So we had to go for that workaround. 

This torch version was the only one that worked for me.

A separate linux requirements would be my preferred way to go as well for now. And we should definitly specify specific package Version Numbers. 

I am still working on the first version, but distributing via pypi and Setup.py is a goal after finishing this.

Reply:
Aaaand I will clean the requirements asap :)